cachelib/cachebench/util/CacheConfig.h

```diff
@@ -41,6 +41,31 @@ class CacheMonitorFactory {
   virtual std::unique_ptr<CacheMonitor> create(Lru2QAllocator& cache) = 0;
 };
 
+// Parse memory tiers configuration from JSON config
+struct MemoryTierConfig : public JSONConfig {
+  MemoryTierConfig() {}
+
+  explicit MemoryTierConfig(const folly::dynamic& configJson);
+  MemoryTierCacheConfig getMemoryTierCacheConfig() {
+    MemoryTierCacheConfig config = memoryTierCacheConfigFromSource();
+    config.setRatio(ratio);
+    config.setMemBind(parseNumaNodes());
+    return config;
+  }
+
+  // Specifies ratio of this memory tier to other tiers
+  size_t ratio{0};
+  // Allocate memory only from specified NUMA nodes
+  std::string memBindNodes{""};
```

Comment:
Can you add comments to clarify what the numa node formats are, and give a few examples?

I don't really understand why they need to look like the following
```
nodes may be specified as N,N,N or N-N or N,N-N or N-N,N-N and so forth
```
E.g. what is N-N and how is it different from N,N?

Reply:
I think it is the same format as used in the libnuma.

I have a question. It seems that we want to eventually get the `struct bitmask` from this node string. Didn't `libnuma` already provide those utility functions like `numa_parse_nodestring()`? Why are we reinventing this parse logic even though we are adding the dependency?

Reply:
@jaesoo-fb good point, I have not noticed that `libnuma` provides a public function to parse `nodestring` and we can use it. 
But now I have another question in mind: Should we pass just `std::string` to the cache config and do parsing inside cachelib or keep the existing interface when we do parsing in `cachebench` and pass `std::vector<size_t>` with node ids to the cache config.
With the first approach `std::string` format is defined by `libnuma`, not by cachelib. Is it OK? what do you think?

Reply:
Well, I think the `memBindNumaNodes` are transparent to `cachelib` except the code in `cachelib/shm` anyway, right? I think we can just passing the node string around and then convert it to `bitmask` inside the `cachelib/shm` when we create the shm.

Reply:
@jaesoo-fb,

I am not sure that accepting NUMA nodes in string format is a good idea for MemoryTierCacheConfig because of the following:

1.  String interface is convenient for CacheBench, but for developers who use C++ cachelib interface it might be not logical to build the string with NUMA nodes config. It is better to have something like `setBit(int n)` interface where each bit corresponds to the NUMA node ID.
2. Another issue is that having a string interface makes us depend on `libnuma` string format. If the string format will be changed in the future (hardly possible, but anyway) it will impact `cachelib` interface.

Based on the abovementioned considerations I have implemented `NumaBitMask` abstraction to encapsulate all logic related to `libnuma`.
Please review the updated patch.

Reply:
I think we don't need to care about the `libnuma` string format change since the chance is low, I believe. However, I like hiding the detail behind the cachelib wrapper`NumaBitMask`.